src/nvim/syntax.c

```diff
@@ -8754,6 +8755,391 @@ RgbValue name_to_color(const char *name)
+{
+  dict_T      *dict;
+  HlGroup     *sgp;
+  dict_T      *attr_dict;
+  int         hlgid;
+
+  dict = tv_dict_alloc();
+  if (dict == NULL) {
+    return dict;
+  }
+
+  sgp = &HL_TABLE()[hl_idx];
+  // highlight group id is 1-based
+  hlgid = hl_idx + 1;
+
+  if (tv_dict_add_str(dict, S_LEN("name"), (const char *)sgp->sg_name) == FAIL) {
+    goto error;
+  }
+  if (tv_dict_add_nr(dict, S_LEN("id"), hlgid) == FAIL) {
+    goto error;
+  }
+
+  if (sgp->sg_link && resolve_link) {
+    // resolve the highlight group link recursively
+    while (sgp->sg_link) {
+      hlgid = sgp->sg_link;
+      sgp = &HL_TABLE()[sgp->sg_link - 1];
+    }
+  }
+
+  if (sgp->sg_cterm != 0) {
+    attr_dict = highlight_get_attr_dict(sgp->sg_cterm);
+    if (attr_dict != NULL) {
+      if (tv_dict_add_dict(dict, S_LEN("cterm"), attr_dict) == FAIL) {
+        goto error;
+      }
+    }
+  }
+  if (sgp->sg_cterm_fg != 0) {
+    if (tv_dict_add_str(dict, S_LEN("ctermfg"),
+                        highlight_color(hlgid, (const char *)"fg", 'c')) == FAIL) {
+      goto error;
+    }
+  }
+  if (sgp->sg_cterm_bg != 0) {
+    if (tv_dict_add_str(dict, S_LEN("ctermbg"),
+                        highlight_color(hlgid, (const char *)"bg", 'c')) == FAIL) {
+      goto error;
+    }
+  }
+  if (sgp->sg_gui != 0) {
+    attr_dict = highlight_get_attr_dict(sgp->sg_gui);
+    if (attr_dict != NULL) {
+      if (tv_dict_add_dict(dict, S_LEN("gui"), attr_dict) == FAIL) {
+        goto error;
+      }
+    }
+  }
+  if (sgp->sg_rgb_fg_name != NULL) {
+    if (tv_dict_add_str(dict, S_LEN("guifg"),
+                        highlight_color(hlgid, (const char *)"fg", 'g')) == FAIL) {
+      goto error;
+    }
+  }
+  if (sgp->sg_rgb_bg_name != NULL) {
+    if (tv_dict_add_str(dict, S_LEN("guibg"),
+                        highlight_color(hlgid, (const char *)"bg", 'g')) == FAIL) {
+      goto error;
+    }
+  }
+  if (sgp->sg_rgb_sp_name != NULL) {
+    if (tv_dict_add_str(dict, S_LEN("guisp"),
+                        highlight_color(hlgid, (const char *)"sp", 'g')) == FAIL) {
+      goto error;
+    }
+  }
+  if (sgp->sg_link) {
+    char_u  *link;
+    link = HL_TABLE()[sgp->sg_link - 1].sg_name;
+    if (link != NULL && tv_dict_add_str(dict, S_LEN("linksto"), (char *)link) == FAIL) {
+      goto error;
+    }
+  }
+  if (dict->dv_hashtab.ht_used == 2) {
+    // If only 'name' is present, then the highlight group is cleared.
+    tv_dict_add_bool(dict, S_LEN("cleared"), kBoolVarTrue);
+  }
+
+  return dict;
+
+error:
+  xfree(dict);
+  return NULL;
+}
+
+// "hlget([name])" function
+// Return the attributes of a specific highlight group (if specified) or all
+// the highlight groups.
+void f_hlget(typval_T *argvars, typval_T *rettv, FunPtr fptr)
+{
+  list_T      *list;
+  dict_T      *dict;
+  int         i;
+  const char  *hlarg = NULL;
+  bool                resolve_link = false;
+
+  tv_list_alloc_ret(rettv, kListLenMayKnow);
+
+  if (argvars[0].v_type != VAR_UNKNOWN) {
```

Comment:
Could have ported the equivalent Vim function in eval/typval.c instead of this.

Reply:
Should I put them part of this patch?  In a separate commit? Or as a different vim-patch (Do you know what vim-path I'll need to port. I'm not sure how to find a patch number TBH :| )?

Reply:
May be <https://github.com/vim/vim/commit/83494b4ac61898f687d6ef9dce4bad5802fb8e51>

Reply:
Thanks @zeertzjq I'll take a look at that patch.

Reply:
@janlazo @zeertzjq most of the changes related seems to be type checking related to vim9. Aren't those NA? Should I just pick the changes in `src/typeval.c` ?

Reply:
Yes.

Reply:
@janlazo is porting those patches in this pr strictly necessary ? Can this be merged without those and those patches ported on a later pr ?